Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,232 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var HighestBalanceEver = decimal.MinValue;
+            var PersonWithHighestBalance = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+
+                if (highestAmountOfPerson > HighestBalanceEver)
+                {
+                    HighestBalanceEver = highestAmountOfPerson;
+                    var currentOne = PersonWithHighestBalance.ToString();
+
+                    if (string.IsNullOrEmpty(currentOne))
+                    {
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
+                    }
+                    else
+                    {
+                        PersonWithHighestBalance.Replace(currentOne, currentPersonData[0]);
+                    }
+
+
+                }
+                else if (highestAmountOfPerson == HighestBalanceEver)
+                {
```

Comment:
Again, the else contents could be moved in a function.

Reply:
Oke, I will try to find out how I can make that work

Reply:
Hmm,  I think I need some hints. Here I change 2 things. The person and the amount.
and if I use a function I can only return 1 thing so I loose data. 

Can you let me see in a example how you see  this ?

Reply:
or do you mean something like this : 

> for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
>             {
>                 var currentPersonData = peopleAndBalances[i].Split(',');
>                 // Find another way to find the max of a Person. 
>                 var isValid = decimal.TryParse(currentPersonData[1..].Max(), out decimal highestCurrentPersonBalance);
> 
>                 if (!isValid)
>                 {
>                     return "N/A.";
>                 }
> 
>                 if (highestCurrentPersonBalance == highestBalanceEver)
>                 {
>                     if (i == peopleAndBalances.Length - 1)
>                     {
>                         personWithHighestBalance.Append(" and ");
>                     }
>                     else
>                     {
>                         personWithHighestBalance.Append(", ");
>                     }
> 
>                     personWithHighestBalance.Append(currentPersonData[0]);
>                 }
> 
>                 highestBalanceEver = CheckIfMorePeopleHaveTheSameAmount(highestBalanceEver, personWithHighestBalance, currentPersonData, highestCurrentPersonBalance);
> 
>             }

Reply:
with a function with 4 arguments

Reply:
I don't know what you mean, you didn't show a function. If you do write a code snippet, make sure it's more relevant.

Reply:
Oke, I will try what I think you mean and make a new commit